### PR TITLE
add locks to transaction related redux actions

### PR DESCRIPTION
**Pull request by estebanmino**
This PR intends to fix the disappearing tx bug adding a lock to all methods that dispatch transactions actions. The main theory we have is that a race condition is happening somewhere I wasn't able to locate, this situation could be updating the transaction state with missing txs and therefore disappearing on the tx list.

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,6 +99,7 @@
     "@walletconnect/client": "1.6.5",
     "@walletconnect/utils": "1.6.5",
     "assert": "1.5.0",
+    "async-mutex": "^0.3.2",
```
Comment (brunobar79) [inline package.json:102]: remove the caret

**Review by brunobar79 — APPROVED** [review]
Code LGTM! 

I'd be very careful with QA and try all types of transactions
=> Send ETH (L1 / L2)
=> Send ERC20s
=> Send NFTs
=> Swaps & Savings Deposits
=> WalletConnect contract interactions
=> Multiple txs (ex. send one with normal and another one with fast before the 1st one confirms)

CC: @skylarbarrera

**Review by jinchung — CHANGES_REQUESTED** [review]
Hey can we follow the pattern that is recommended by the `async-mutex` library?
Namely:
1. Making sure there is a try/finally block around the code of interest to enforce release of the lock (otherwise, there could be an error thrown and we deadlock)
2. The current pattern uses the manual acquire and release which has risks in development (forgetting to do the "release" as we move code around) - vs using the `runExclusive`

**Comment by estebanmino** [comment]
@jinchung @brunobar79 updated with the suggested changes!

**Comment by skylarbarrera** [comment]
looks like this is messing with L2 transaction approvals, these transactions were confirmed for >10 min

https://user-images.githubusercontent.com/29204161/134047622-1beeac31-c00c-4f0b-baee-06b728ba8a78.mov

**Comment by estebanmino** [comment]
@skylarbarrera nice catch! I think it was because the provider was defaulting to the web3Provider which is on mainnet so the tx couldn't be found, I could repro and is fixed now

**Review by skylarbarrera — APPROVED** [review]
💯

File: package.json (hunk 1 above)
Comment (mikedemarais) [reply package.json]: the caret nazi strikes again

**Review by jinchung — APPROVED** [review]
🌮 

approve pending confirmation of `network` always existing on pending txn

**Comment by jinchung** [comment]
Since NewTransaction type technically has an optional network value, then we can default the `getProviderForNetwork` to using `mainnet` @brunobar79 

Also, we can move towards using `getProviderForNetwork` instead of `web3Provider` fallbacks in places to have a safer pattern so devs don't need to "remember" this pattern

File: package.json (hunk 1 above)
Comment (estebanmino) [reply package.json]: lol

File: src/parsers/transactions.ts [hunk 2]
```diff
@@ -364,6 +365,7 @@ const parseTransaction = (
         minedAt: txn.mined_at,
         name: updatedAsset.name,
         native: nativeDisplay,
+        network,
```
Comment (jinchung) [inline src/parsers/transactions.ts:368]: 🤔 hmmm

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**